From 8266cbcf4563011b3a6686405f2fd56964123506 Mon Sep 17 00:00:00 2001
From: Jan Kotas <jkotas@microsoft.com>
Date: Wed, 10 Jan 2018 22:06:28 -0800
Subject: [PATCH] Fix Obsolete on ReadOnlySpan Equals and GetHashCode methods
 (#15820)

* Fix Obsolete on ReadOnlySpan Equals and GetHashCode methods

* Delete unnecessary CLSCompliant attribute
---
 src/mscorlib/shared/System/ReadOnlySpan.cs | 4 ++--
 src/mscorlib/src/System/RuntimeHandles.cs  | 1 -
 2 files changed, 2 insertions(+), 3 deletions(-)

diff --git a/src/mscorlib/shared/System/ReadOnlySpan.cs b/src/mscorlib/shared/System/ReadOnlySpan.cs
index dac8069..1854df4 100644
--- a/src/mscorlib/shared/System/ReadOnlySpan.cs
+++ b/src/mscorlib/shared/System/ReadOnlySpan.cs
@@ -240,7 +240,7 @@ namespace System
         /// Always thrown by this method.
         /// </exception>
         /// </summary>
-        [Obsolete("Equals() on Span will always throw an exception. Use == instead.")]
+        [Obsolete("Equals() on ReadOnlySpan will always throw an exception. Use == instead.")]
         [EditorBrowsable(EditorBrowsableState.Never)]
         public override bool Equals(object obj)
         {
@@ -253,7 +253,7 @@ namespace System
         /// Always thrown by this method.
         /// </exception>
         /// </summary>
-        [Obsolete("GetHashCode() on Span will always throw an exception.")]
+        [Obsolete("GetHashCode() on ReadOnlySpan will always throw an exception.")]
         [EditorBrowsable(EditorBrowsableState.Never)]
         public override int GetHashCode()
         {
diff --git a/src/mscorlib/src/System/RuntimeHandles.cs b/src/mscorlib/src/System/RuntimeHandles.cs
index 69ee4c2..dc84976 100644
--- a/src/mscorlib/src/System/RuntimeHandles.cs
+++ b/src/mscorlib/src/System/RuntimeHandles.cs
@@ -244,7 +244,6 @@ namespace System
         [MethodImplAttribute(MethodImplOptions.InternalCall)]
         internal extern static RuntimeModule GetModule(RuntimeType type);
 
-        [CLSCompliant(false)]
         public ModuleHandle GetModuleHandle()
         {
             return new ModuleHandle(RuntimeTypeHandle.GetModule(m_type));
-- 
2.7.4

